### PR TITLE
[WIP] clean-up: remove cluster provisioning from e2e

### PR DESCRIPTION
**Pull request by perdasilva**
**Description of the change:**
* Removes cluster provisioning from the e2e test suite and moves that responsibility to the Makefile
* Adds bingo to the project to help manage the different tools needed for the project (e.g. kind, setup_envtest, helm, etc.)

**Motivation for the change:**

The project is currently a bit messy:
 - e2e suite doing things it shouldn't worry about
 - Makefile with stuff many don't understand/use
 - different build tags
 - scripts that could have been a couple of lines in the Makefile

So, trying to clean it up to make it easier to maintain

**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,19 +24,30 @@ IMAGE_REPO := quay.io/operator-framework/olm
...
 ARCH := arm64
 else
 ARCH := 386
 endif
+
+KIND_CLUSTER_NAME ?= kind-olmv0
+# Not guaranteed to have patch releases available and node image tags are full versions (i.e v1.28.0 - no v1.28, v1.29, etc.)
+# The KIND_NODE_VERSION is set by getting the version of the k8s.io/client-go dependency from the go.mod
+# and sets major version to "1" and the patch version to "0". For example, a client-go version of v0.28.5
+# will map to a KIND_NODE_VERSION of 1.28.0
```
Comment (kevinrizza) [inline Makefile:47]: Thank you for adding comments!

**Review by kevinrizza — COMMENTED** [review]
I took a quick look and I liked what I saw. Can you update the readme in the root of the project to describe the expected developer workflow now that we're removing the `deploy-local` target from the makefile?

File: Makefile (hunk 1 above)
Comment (perdasilva) [reply Makefile:47]: I took a lot of this from controller-operator - so I deserve no praise here!

File: Makefile [hunk 2]
```diff
@@ -21,19 +24,30 @@ IMAGE_REPO := quay.io/operator-framework/olm
...
+KIND_NODE_VERSION = $(shell go list -m k8s.io/client-go | cut -d" " -f2 | sed 's/^v0\.\([[:digit:]]\{1,\}\)\.[[:digit:]]\{1,\}$$/1.\1.0/')
```
Comment (joelanford) [inline Makefile:48]: Example of a change that I am suggesting in https://github.com/operator-framework/operator-lifecycle-manager/issues/3215

But fine to leave that out-of-scope of this PR if we want to tackle that all at the same time.
```suggestion
KIND_NODE_VERSION := $(shell go list -m k8s.io/client-go | cut -d" " -f2 | sed 's/^v0\.\([[:digit:]]\{1,\}\)\.[[:digit:]]\{1,\}$$/1.\1.0/')
```

File: Makefile [hunk 3]
```diff
@@ -53,7 +67,7 @@ unit: kubebuilder
 KUBEBUILDER_ASSETS_ERR := not detected in $(KUBEBUILDER_ASSETS), to override the assets path set the KUBEBUILDER_ASSETS environment variable, for install instructions see https://pkg.go.dev/sigs.k8s.io/controller-runtime/tools/setup-envtest
```
Comment (joelanford) [inline Makefile:67]: It would be great to change this to do a similar automatic setup from `setup-envtest`  like we do in operator-controller.

Comment (perdasilva) [reply Makefile:67]: Yup yup - already have that in some commit in here somewhere. Actually, I think I might remove bingo from this PR, then add it as its own. To keep this change even smaller.

**Comment by perdasilva** [comment]
closing in favor of #3222

**Pull request closed without merging**